lib/sdk11/components/libraries/bootloader_dfu/dfu_transport_ble.c

```diff
@@ -1088,3 +1094,109 @@ uint32_t dfu_transport_ble_close()
 
     return NRF_SUCCESS;
 }
+
+#ifdef CFG_DEBUG
+
+typedef struct
+{
+  uint32_t key;
+  void const * data;
+}lookup_entry_t;
+
+typedef struct
+{
+  uint16_t count;
+  lookup_entry_t const* items;
+} lookup_table_t;
+
+void const * lookup_find(lookup_table_t const* p_table, uint32_t key)
+{
+  for(uint16_t i=0; i<p_table->count; i++)
+  {
+    if (p_table->items[i].key == key) return p_table->items[i].data;
+  }
+
+  return NULL;
+}
+
+/*------------------------------------------------------------------*/
+/* BLE Event String
+ *------------------------------------------------------------------*/
+static lookup_entry_t const _strevt_lookup[] =
+{
+    // BLE common: 0x01
+    { .key = BLE_EVT_USER_MEM_REQUEST                , .data= "BLE_EVT_USER_MEM_REQUEST"                },

```

Comment:
You could probably do this with a preprocessor macro, so you avoid typing the name twice (and possibly introducing a typo). But not a big deal.

Reply:
ah yeah, you are right, I will do that later on :)